src/main/java/net/coderbot/iris/gl/shader/StandardMacros.java

```diff
@@ -0,0 +1,202 @@
+
+
+	/**
+	 * Gets the current mc version String in a 5 digit format
+	 *
+	 * @return mc version string
+	 * @see <a href="https://github.com/sp614x/optifine/blob/9c6a5b5326558ccc57c6490b66b3be3b2dc8cbef/OptiFineDoc/doc/shaders.txt#L696-L699">Optifine Doc</a>
+	 */
+	public static String getMcVersion() {
+		String version = SharedConstants.getGameVersion().getReleaseTarget(); //release target so snapshots are set to the higher version
+
+		if (version == null) {
+			throw new IllegalStateException("Could not get the current minecraft version!");
+		}
+
+		String[] splitVersion = version.split("\\.");
+
+		if (splitVersion.length < 2) throw new IllegalStateException("Could not parse game version into a 5 digit format!");
+
+		String major = splitVersion[0];
+		String minor = splitVersion[1];
+		String bugfix;
+
+		if (splitVersion.length < 3) {
+			bugfix = "00";
+		} else {
+			bugfix = splitVersion[2];
+		}
+
+		if (minor.length() == 1) {
+			minor = 0 + minor;
+		}
+		if (bugfix.length() == 1) {
+			bugfix = 0 + bugfix;
+		}
+
+		return major + minor + bugfix;
+	}
+
+	/**
+	 * Returns the current OS String
+	 *
+	 * @return the string based on the current OS
+	 * @see <a href="https://github.com/sp614x/optifine/blob/9c6a5b5326558ccc57c6490b66b3be3b2dc8cbef/OptiFineDoc/doc/shaders.txt#L709-L714">Optifine Doc</a>
+	 */
+	public static String getOsString() {
+		switch (Util.getOperatingSystem()) {
+			case OSX:
+				return "MC_OS_MAC";
+			case LINUX:
+				return "MC_OS_LINUX";
+			case WINDOWS:
+				return "MC_OS_WINDOWS";
+			case SOLARIS: //Note: Optifine doesn't have a macro for Solaris. https://github.com/sp614x/optifine/blob/master/OptiFineDoc/doc/shaders.txt#L689-L692
+			case UNKNOWN:
+			default:
+				return "MC_OS_UNKNOWN";
+		}
+	}
+
+	/**
+	 * Returns the current GL Version using regex
+	 *
+	 * @param name the name of the gl attribute to parse
+	 * @return current gl version stripped of semantic versioning
+	 * @see <a href="https://github.com/sp614x/optifine/blob/9c6a5b5326558ccc57c6490b66b3be3b2dc8cbef/OptiFineDoc/doc/shaders.txt#L701-L703">Optifine Doc for GL Version</a>
+	 * @see <a href="https://github.com/sp614x/optifine/blob/9c6a5b5326558ccc57c6490b66b3be3b2dc8cbef/OptiFineDoc/doc/shaders.txt#L705-L707">Optifine Doc for GLSL Version</a>
+	 */
+	public static String getGlVersion(int name) {
+		String info = GL20.glGetString(name);
+		System.out.println("GL Version: " + info);
+
+		Matcher matcher = SEMVER_PATTERN.matcher(Objects.requireNonNull(info));
+
+		if (!matcher.matches()) {
+			throw new IllegalStateException("Could not parse GL version into 5 digit format");
+		}
+
+		String major = group(matcher, "major");
+		String minor = group(matcher, "minor");
+		String bugfix = group(matcher, "bugfix");
+
+		if (bugfix == null) {
+			//if bugfix is not there, it is 0
+			bugfix = "0";
+		}
+
+		if (major == null || minor == null) throw new IllegalStateException("Could not parse GL Version into 5 digit format!");
```

Comment:
Similar thing to before
```suggestion
		if (major == null || minor == null) {
			throw new IllegalStateException("Could not parse GL version " + info + "!");
		}
```